### PR TITLE
update main README file

### PR DESCRIPTION
**Pull request by nickmoreton**

### CODE REVIEW

**Comment by nickmoreton** [comment]
@nimasmi I just realised the links to the other pages aren't set correctly. I think they don't need the package folder name. I will change them and resubmit it.

**Comment by nickmoreton** [comment]
Links are fixed now.

File: README.md [hunk 1]
```diff
@@ -1,77 +1,172 @@
...
+
+1. Wagtail CMS Installed with initial setup
+2. Wordpress XML export of all content in a single file
+3. The Wordpress website will need to be live and available for importing of assets such as images and documents.
+
+## Initial app and package setup
+
+1. Setup a Wagtail site using your preferred method or follow the [official documentation](https://docs.wagtail.io/en/stable/getting_started/tutorial.html) to get started.
+2. Download the repo for this package [Wagtaill Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)
+3. Install this package with pip install -e path/to/wagtail-wordpress-import or using any method you prefer.
```
Comment (nimasmi) [inline README.md]: **Question**: Why can't this be installed with `pip install -e git+https://github.com/torchbox/wagtail-wordpress-import/`, i.e. without downloading?

**Suggestion**: Fix typo.

```suggestion
2. Download the repo for this package [Wagtail Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)
3. Install this package with pip install -e path/to/wagtail-wordpress-import or using any method you prefer.
```

File: README.md [hunk 2]
```diff
@@ -1,77 +1,172 @@
...
+from wagtail_wordpress_import.models import WPImportedPageMixin
+class PostPage(WPImportedPageMixin, Page):
+    ...
+```
 
-Place your XML files somewhere on your disk. The file can have any name you choose.
+You will need to run `python manage.py makemigrations` and `python manage.py migrate` to add the fields to your page model.
 
-# Import XML
+*It's intended that this initial setup can be removed once the content has been imported. [view source](wagtail_wordpress_import/models.py)*
```
Comment (nimasmi) [inline README.md]:
```suggestion
*It's intended that these fields are temporary for while importing, and can be removed once the content has been imported. [view source](wagtail_wordpress_import/models.py)*
```

File: README.md [hunk 3]
```diff
@@ -1,77 +1,172 @@
...
 
-### Default import command
+        # debug fields
+        self.wp_post_id = data["wp_post_id"]
+        self.wp_post_type = data["wp_post_type"]
+        self.wp_link = data["wp_link"]
+        self.wp_raw_content = data["wp_raw_content"]
+        self.wp_block_json = data["wp_block_json"]
+        self.wp_processed_content = data["wp_processed_content"]
+        self.wp_normalized_styles = data["wp_normalized_styles"]
```
Comment (nimasmi) [inline README.md:95]: **Comment**: It seem odd to expect the implementer to declare all these, when they're inheriting from a supplied base class. Can't we have the Wagtail Page model fields, and the debug fields, in a parent class method somewhere, and require the implementer to specify only the body field below?

File: README.md [hunk 4]
```diff
@@ -1,77 +1,172 @@
...
+3. Install this package with pip install -e path/to/wagtail-wordpress-import or using any method you prefer.
+4. Place your XML files somewhere on your disk. The file can have any name you choose.
 
-## Installation
+### First steps to configure your wagtail app
 
-* Install the package with `pip install wagtail-wordpress-import` (not yet available) but can be installed with pip install -e wagtail-wordpress-import for development.
+The import can be run on an existing or new site but you will need to perform some setup on your page models.
 
+We recommend your page model inherits from the provided WPImportedPageMixin
```
Comment (nimasmi) [inline README.md:37]: **Suggestion**: It must declare an `import_wordpress_data` method.

However, I'm not sure how I feel about the implementation, see my comment on that method below.

File: README.md [hunk 5]
```diff
@@ -1,77 +1,172 @@
...
+### Optional command arguments
 
-[View Documentation](docs/mapping.md)
+- `-m` can be used to specify the Wagtail Page model to use for all imported pages. The default is `PostPage` when it's not specified.
+- `-a` can be used to specify the Wagtail App where you have created your Wagtail Page model. The default is `pages` when it's not specified.
+- `-t` can be used to limit the Wordpress page types to be imported. You can pass in a comma separated string of page types or just a single page type. The default is `page,post` if not specified.
+- `-s` can be used to specify the status of pages you want to import. You can pass in a comma separated string of statuses or just a single status. The default is `publish,draft` if not specified.
 
-# Creating page models
+[View the import command source](wagtail_wordpress_import/management/commands/import_xml.py) to if you need to extend the command.
```
Comment (nimasmi) [inline README.md]: **Issue**: I'm not sure what this sentence is intended to say.

File: README.md [hunk 6]
```diff
@@ -1,77 +1,172 @@
...
+Image.objects.all().delete()
+```
+
+#### Delete all imported documents
+
+```shell
+from wagtail.documents.models import Document
++
+Document.objects.all().delete()
+```
```
Comment (nimasmi) [inline README.md]: **Issue**: These commands both look like they will delete _all_ images and documents.

File: README.md [hunk 7]
```diff
@@ -1,77 +1,172 @@
...
 
-- [import_xml.py](wagtail_wordpress_import/management/commands/import_xml.py)
+## Running the import command
 
-The configuration is set to import all items in the XML file that have a `<wp:post_type>` = `post` or `page`. The command only imports an item that has `<wp:status>` = `draft` or `publish`. By default each item is created using a Page model of PostPage()
+The most basic command would be:
 
-## Changing the import configuration
+```bash
+`python manage.py import_xml path/to/xml/file.xml parent_page_id`
```
Comment (nimasmi) [inline README.md]: **Suggestion**: Remove backticks inside code block.

```suggestion
python manage.py import_xml path/to/xml/file.xml parent_page_id
```

File: README.md (hunk 1 above)
Comment (nickmoreton) [reply README.md]: > Why can't this be installed with `pip install -e git+https://github.com/torchbox/wagtail-wordpress-import/`, i.e. without downloading?

That makes sense but I see:
```
ERROR: Could not detect requirement name for 'git+https://github.com/torchbox/wagtail-wordpress-import/', please specify one with #egg=your_package_name
```

**Comment by nickmoreton** [comment]
Hi @nimasmi , I realise there's a couple of steps I missed in the initial setup, when I tried a fresh install earlier, is it ok if I add them in now.

File: README.md (hunk 1 above)
Comment (nimasmi) [reply README.md]: Ah, yes, sorry. For me locally it works with

```
pip install -e "git+https://github.com/torchbox/wagtail-wordpress-import.git#egg=wagtail-wordpress-import"
```

**Comment by nimasmi** [comment]
> Hi @nimasmi , I realise there's a couple of steps I missed in the initial setup, when I tried a fresh install earlier, is it ok if I add them in now.

Yep, just shout when you're ready.

File: README.md (hunk 5 above)
Comment (nickmoreton) [reply README.md]: I think one of my just in case suggestions again. Will remove it.

File: README.md (hunk 3 above)
Comment (nickmoreton) [reply README.md:95]: I have created a ticket to alter this: https://projects.torchbox.com/projects/wordpress-to-wagtail-importer-package/tickets/85

File: README.md (hunk 4 above)
Comment (nickmoreton) [reply README.md:37]: New ticket created to alter this: https://projects.torchbox.com/projects/wordpress-to-wagtail-importer-package/tickets/85

File: README.md [hunk 8]
```diff
@@ -23,9 +24,12 @@ A package for Wagtail CMS to import WordPress blog content from an XML file into
 ## Initial app and package setup
 
 1. Setup a Wagtail site using your preferred method or follow the [official documentation](https://docs.wagtail.io/en/stable/getting_started/tutorial.html) to get started.
-2. Download the repo for this package [Wagtaill Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)
-3. Install this package with pip install -e path/to/wagtail-wordpress-import or using any method you prefer.
+2. Download the repo for this package [Wagtail Wordpress Import](https://github.com/torchbox/wagtail-wordpress-import) to a location on your hard-drive. (This is the latest development branch)
+3. Install this package with pip install -e "git+https://github.com/torchbox/wagtail-wordpress-import.git#egg=wagtail-wordpress-import"
+ or using any method you prefer.
```
Comment (nimasmi) [inline README.md]: **Issue**: Now we're telling people to install from GitHub, they won't need the download step.

Comment (nickmoreton) [reply README.md]: Of course :) I removed it.

**Review by nimasmi — APPROVED** [review]

**Pull request merged**